lib/server.js

```diff
@@ -926,6 +926,24 @@ WalletService.prototype.getMainAddresses = function(opts, cb) {
 };
 
 /**
+ * Get a specific address object from a specific wallet.
+ * @param {Object} opts
+ * @param {String} opts.address - The address.
+ * @param {String} opts.walletId - The wallet id.
```

Comment:
`walletId` should be taken from `self.walletId` which is automatically set to the current user's wallet. Otherwise, we are allowing anyone to query all wallets.